x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/esnative/NativeRealmIntegTests.java

```diff
@@ -269,7 +270,24 @@ public void testAddAndGetRole() throws Exception {
         assertFalse("role should not exist after being deleted", resp2.hasRoles());
     }
 
-    public void testAddUserAndRoleThenAuth() throws Exception {
+    public void testAddUserAndRoleThenAuth() {
+        testAddUserAndRoleThenAuth("joe");
+    }
+
+    public void testAddUserWithInternalUsernameAndRoleThenAuth() {
+        testAddUserAndRoleThenAuth(randomInternalUsername());
+    }
+
+    public void testAuthWithInternalUsernameFailsWithoutCorrespondingUser() {
```

Comment:
Maybe overly paranoid. LMK if I should drop this since it's quite heavy

Reply:
I like it.